sympy/core/function.py

```diff
@@ -909,6 +919,7 @@ def __new__(mcl, name, bases=(AppliedUndef,), __dict__=None, **kwargs):
         __dict__['__module__'] = None
         obj = super(UndefinedFunction, mcl).__new__(mcl, name, bases, __dict__)
         obj.name = name
+        obj._sage_ = UndefSage()
```

Comment:
Maybe I misunderstood but it looks as if we can used a single instance of `UndefSage` for all classes rather than creating an instance for each UndefinedFunction class.